dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java

```diff
@@ -109,6 +111,47 @@ public final Response shutdown(@Context final HttpServletRequest request,
         return Response.ok(new ResponseEntityView("Shutdown")).build();
     }
 
+    /**
+     * This method is meant to shut down the current DotCMS instance.
+     * It will pass the control to catalina.sh (Tomcat) script to deal with any exit code.
+     * 
+     * @param request http request
+     * @param response http response
+     * @return string response
+     */
+    @DELETE
+    @Path("/_shutdownCluster")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response shutdownCluster(@Context final HttpServletRequest request,
+                                   @Context final HttpServletResponse response,
+                                   @DefaultValue("60") @QueryParam("rollingDelay") int rollingDelay) {
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredRoles(Role.CMS_ADMINISTRATOR_ROLE)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .requiredPortlet("maintenance")
+                .init();
+
+
+        Logger.info(this.getClass(), "User:" + initData.getUser() + " is shutting down dotCMS Cluster with a rolling delay of " + rollingDelay); 
+        SecurityLogger.logInfo(
+                this.getClass(),
+                "User:" + initData.getUser() + " is shutting down dotCMS Cluster with a rolling delay of " + rollingDelay);
+
+        if (!Config.getBooleanProperty("ALLOW_DOTCMS_SHUTDOWN_FROM_CONSOLE", true)) {
```

Comment:
We should not use Config directly, use Lazy instead to avoid blocking issues

Reply:
Done!